woocommerce/woocommerce
# Automate Changelog: Create CLI command, display PR info

### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:



Working towards https://github.com/woocommerce/woocommerce/issues/38125

In order to automate a changelog entry from PR description, the first step will be to extract data from the PR. This PR creates a CLI command to extract data from any pull request.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. See the "Changelog entry" section below. This will be added to all pull requests as part of a the pull request template when this feature is ready. Does it make sense? Any suggestions?
2. Run the command on this pull request. The data will be printed in the CLI command output.
```
pnpm utils changelog 38327
```
3. Try editing this PR description and see the printed data update. Note that I'm not validating the data at this point. This will be done by Jetpack Changelogger in a next step.
4. Run the command again with a community PR that has the entry data, like this one: https://github.com/woocommerce/woocommerce/pull/38267. The `isCommunityPR` field should be correct.




### Changelog entry




<details>

#### Significance


#### Type


#### Message

This is a very useful fix.

#### Comment


</details>
